gdnative-core/src/geom/basis.rs

```diff
@@ -29,14 +116,316 @@ impl Basis {
+                }
+            };
+
+            let j = (i + 1) % 3;
+            let k = (i + 2) % 3;
+
+            let elements_arr: [[f32; 3]; 3] = [
+                self.elements[0].to_array(),
+                self.elements[1].to_array(),
+                self.elements[2].to_array(),
+            ];
+
+            let mut s = (elements_arr[i][i] - elements_arr[j][j] - elements_arr[k][k] + 1.0).sqrt();
+            temp[i] = s * 0.5;
+            s = 0.5 / s;
+
+            temp[3] = (elements_arr[k][j] - elements_arr[j][k]) * s;
+            temp[j] = (elements_arr[j][i] + elements_arr[i][j]) * s;
+            temp[k] = (elements_arr[k][i] + elements_arr[i][k]) * s;
         }
+
+        let [a, b, c, r] = temp;
+        Quat::quaternion(a, b, c, r)
+    }
+
+    /// Converts matrix into a Quaternion
+    ///
+    /// See [`Basis::into_quat`](./struct.Basis.html#method.into_quat)
+    pub fn quat(&self) -> Quat {
+        // Assumes that the matrix can be decomposed into a proper rotation and scaling matrix as M = R.S,
+        // and returns the Euler angles corresponding to the rotation part, complementing get_scale().
+        // See the comment in get_scale() for further information.
+        let mut m = self.orthonormalized();
+        let det = m.determinant();
+        if det < 0.0 {
+            // Ensure that the determinant is 1, such that result is a proper rotation matrix which can be represented by Euler angles.
+            m.set_scale(&Vector3::new(-1.0, -1.0, -1.0));
+        }
+
+        return m.into_quat();
+    }
+
+    /// Returns the scale of the matrix.
+    pub fn scale(&self) -> Vector3 {
+        let det = self.determinant();
+        let det_sign = if det < 0.0 { -1.0 } else { 1.0 };
+        Vector3::new(
+            Vector3::new(self.elements[0].x, self.elements[1].x, self.elements[2].x).length(),
+            Vector3::new(self.elements[0].y, self.elements[1].y, self.elements[2].y).length(),
+            Vector3::new(self.elements[0].z, self.elements[1].z, self.elements[2].z).length(),
+        ) * det_sign
+    }
+
+    /// Multiplies the matrix from left by the scaling matrix: M -> S.M
+    /// See the comment for Basis::rotated for further explanation.
+    fn set_scale(&mut self, s: &Vector3) {
+        self.elements[0] *= s.x;
+        self.elements[1] *= s.y;
+        self.elements[2] *= s.z;
+    }
+
+    /// Introduce an additional scaling specified by the given 3D scaling factor.
+    pub fn scaled(mut self, scale: &Vector3) -> Basis {
+        self.set_scale(scale);
+        self
+    }
+
+    /// Returns the basis’s rotation in the form of Euler angles (in the YXZ convention: first Z, then X, and Y last).
+    /// The returned vector contains the rotation angles in the format (X angle, Y angle, Z angle).
+    /// See get_rotation_quat if you need a quaternion instead.
+    pub fn euler(&self) -> Vector3 {
+        self.euler_yxz()
+    }
+
+    fn euler_yxz(&self) -> Vector3 {
```

Comment:
The contents of this method should just be in `euler` (or `to_euler` after the rename).